common/workunit/workunit.cpp

```diff
@@ -8721,6 +8725,65 @@ void CLocalWorkUnit::setDebugValue(const char *propname, const char *value, bool
     }
 }
 
+static const char *summaryTypeName(SummaryType type)
+{
+    switch (type)
+    {
+    case SummaryType::ReadFile: return "ReadFile";
+    case SummaryType::ReadIndex: return "ReadIndex";
+    case SummaryType::WriteFile: return "WriteFile";
+    case SummaryType::WriteIndex: return "WriteIndex";
+    case SummaryType::PersistFile: return "PersistFile";
+    case SummaryType::SpillFile: return "SpillFile";
+    case SummaryType::JobTemp: return "JobTemp";
+    case SummaryType::Service: return "Service";
+    default:
+        throwUnexpected();
+    }
+};
+
+bool CLocalWorkUnit::getSummary(SummaryType type, SummaryMap &map) const
+{
+    VStringBuffer xpath("Summaries/%s", summaryTypeName(type));
+    CriticalBlock block(crit);
+    const char *list = p->queryProp(xpath);
+    if (!list)
+        return false;
+    StringArray s;
+    s.appendList(list, "\n");
+    ForEachItemIn(idx, s)
+    {
+        const char *name = s.item(idx);
+        if (name && *name)
+        {
+            char *end = nullptr;
+            SummaryFlags flags = (SummaryFlags) strtol(name, &end, 16);
+            if (*end!=':')
+                return false; // unrecognized format
+            name = end+1;
+            if (map.find(name) == map.end())
+                map[name] = flags;
+            else
+                map[name] = map[name] & flags;
```

Comment:
trivial: could be:
```
            auto it = map.find(name);
            if (it == map.end())
                map[name] = flags;
            else
                it->second &= flags;
```

Reply:
yes, I will fix in a follow on PR (along with the similar code below)